src/object/srv_obj_migrate.c

```diff
@@ -2210,19 +2235,21 @@ migrate_enum_unpack_cb(struct dc_obj_enum_unpack_io *io, void *data)
 	if (tls == NULL || tls->mpt_fini) {
 		D_WARN("some one abort the rebuild "DF_UUID"\n",
 		       DP_UUID(arg->arg->pool_uuid));
-		return 0;
+		D_GOTO(put, rc = 0);
 	}
 
-	if (!arg->ec_rotate_parity)
-		io->ui_dkey_hash = 0;
-
+	migrate_tgt_off = obj_ec_shard_off_by_layout_ver(arg->new_layout_ver, io->ui_dkey_hash,
+							 &arg->oc_attr, shard);
+	unpack_tgt_off = obj_ec_shard_off(obj_hdl2ptr(arg->oh), io->ui_dkey_hash,
+					  io->ui_oid.id_shard);
```

Comment:
sorry, what's the difference between "shard" and "io->ui_oid.id_shard"? a few comment might be helpful, thx

Reply:
Ah, id_shard is the fetched shard, i.e. where do we get the data,  shard is the shard we will "rebuild".